src/platform/webos/PlatformManagerImpl.cpp

```diff
@@ -0,0 +1,581 @@
+        }
+    }
+    else
+    {
+        ChipLogError(DeviceLayer, "Failed to get current uptime since the Node’s last reboot");
+    }
+
+    return Internal::GenericPlatformManagerImpl_POSIX<PlatformManagerImpl>::_Shutdown();
+}
+
+CHIP_ERROR PlatformManagerImpl::_GetFixedLabelList(
+    EndpointId endpoint, AttributeList<app::Clusters::FixedLabel::Structs::LabelStruct::Type, kMaxFixedLabels> & labelList)
+{
+    // In Linux simulation, return following hardcoded labelList on all endpoints.
+    FixedLabel::Structs::LabelStruct::Type room;
+    FixedLabel::Structs::LabelStruct::Type orientation;
+    FixedLabel::Structs::LabelStruct::Type floor;
+    FixedLabel::Structs::LabelStruct::Type direction;
+
+    room.label = CharSpan::fromCharString("room");
+    room.value = CharSpan::fromCharString("bedroom 2");
+
+    orientation.label = CharSpan::fromCharString("orientation");
+    orientation.value = CharSpan::fromCharString("North");
+
+    floor.label = CharSpan::fromCharString("floor");
+    floor.value = CharSpan::fromCharString("2");
+
+    direction.label = CharSpan::fromCharString("direction");
+    direction.value = CharSpan::fromCharString("up");
+
+    labelList.add(room);
+    labelList.add(orientation);
+    labelList.add(floor);
+    labelList.add(direction);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR
+PlatformManagerImpl::_SetUserLabelList(
+    EndpointId endpoint, AttributeList<app::Clusters::UserLabel::Structs::LabelStruct::Type, kMaxUserLabels> & labelList)
+{
+    // TODO:: store the user labelList, and read back stored user labelList if it has been set. Add yaml test to verify this.
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR
+PlatformManagerImpl::_GetUserLabelList(
+    EndpointId endpoint, AttributeList<app::Clusters::UserLabel::Structs::LabelStruct::Type, kMaxUserLabels> & labelList)
+{
+    // In Linux simulation, return following hardcoded labelList on all endpoints.
+    UserLabel::Structs::LabelStruct::Type room;
+    UserLabel::Structs::LabelStruct::Type orientation;
+    UserLabel::Structs::LabelStruct::Type floor;
+    UserLabel::Structs::LabelStruct::Type direction;
+
+    room.label = CharSpan::fromCharString("room");
+    room.value = CharSpan::fromCharString("bedroom 2");
+
+    orientation.label = CharSpan::fromCharString("orientation");
+    orientation.value = CharSpan::fromCharString("North");
+
+    floor.label = CharSpan::fromCharString("floor");
+    floor.value = CharSpan::fromCharString("2");
+
+    direction.label = CharSpan::fromCharString("direction");
+    direction.value = CharSpan::fromCharString("up");
+
+    labelList.add(room);
+    labelList.add(orientation);
+    labelList.add(floor);
+    labelList.add(direction);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR
+PlatformManagerImpl::_GetSupportedLocales(AttributeList<chip::CharSpan, kMaxLanguageTags> & supportedLocales)
+{
+    // In Linux simulation, return following hardcoded list of Strings that are valid values for the ActiveLocale.
```

Comment:
this is webos, should it reference linux?

Reply:
Yes, it referenced linux side and we verified it on webOS platform.